### PR TITLE
[NFC][DF] Improve documentation for Vary()

### PR DESCRIPTION
**Pull request by martamaja10**
An attempt to make Vary() documentation more clear and consistent (given we have 11 overloads) and up to date, as requested on the forum. 

I built the website locally to check if everything looks ok.

### CODE REVIEW

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 1]
```diff
@@ -695,10 +700,32 @@ public:
...
+   /// \param[in] colName set of names of columns for which varied values are provided.
+   /// \param[in] expression a callable that evaluates the varied values for the specified columns. The callable can
+   ///            take any column values as input, similarly to what happens during Filter and Define calls. It must
+   ///            return an RVec of varied values, one for each variation tag, in the same order as the tags.
+   /// \param[in] inputColumns the names of the columns to be passed to the callable.
+   /// \param[in] nVariations number of variations returned by the expression. The corresponding tags will be "0", "1",
+   /// etc. \param[in] variationName a generic name for this set of varied values, e.g. "ptvariation".
+   ///            colName is used if none is provided.
+   ///
+   /// This overload of Vary takes an nVariations parameter instead of a list of tag names. Tag names
```
Comment (vepadulano) [inline tree/dataframe/inc/ROOT/RDF/RInterface.hxx:713]: Maybe add another bit of explanation about the `variationName` (very rough probably can be improved):
"The varied results will be accessible via the keys of the dictionary with the form `variationName:N` where N is the corresponding sequential tag up to `nVariations`"

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 2]
```diff
@@ -738,7 +782,20 @@ public:
       return VaryImpl<false>(colNames, std::forward<F>(expression), inputColumns, variationTags, variationName);
    }
 
-   /// Overload to avoid ambiguity between C++20 string, vector<string> construction from init list.
+   /// \brief Register systematic variations for multiple existing columns using custom variation tags.
+   /// \param[in] colNames name of the column for which varied values are provided.
```
Comment (vepadulano) [inline tree/dataframe/inc/ROOT/RDF/RInterface.hxx:786]: Probably typo
```suggestion
   /// \param[in] colNames set of names of columns for which varied values are provided.
```

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 3]
```diff
@@ -851,30 +976,46 @@ public:
       return Vary(colNames, expression, std::move(variationTags), variationName);
    }
 
-   /// Overload to avoid ambiguity between C++20 string, vector<string> construction from init list.
+   /// \brief Register systematic variations for multiple existing columns using auto-generated variation tags.
+   /// \param[in] colNames name of the column for which varied values are provided.
```
Comment (vepadulano) [inline tree/dataframe/inc/ROOT/RDF/RInterface.hxx:980]: Probably typo
```suggestion
   /// \param[in] colNames set of names of columns for which varied values are provided.
```

File: tree/dataframe/src/RDataFrame.cxx [hunk 4]
```diff
@@ -1059,12 +1059,15 @@ all_hs.GetKeys(); // returns {"nominal", "pt:down", "pt:up", "eta:0", "eta:1"}
 Note how we passed the integer `2` instead of a list of variation tags to the second Vary() invocation: this is a
 shorthand that automatically generates tags 0 to N-1 (in this case 0 and 1).
 
-\note As of v6.26, VariationsFor() and RResultMap are in the `ROOT::RDF::Experimental` namespace, to indicate that these
+\note As of v6.30, VariationsFor() and RResultMap are in the `ROOT::RDF::Experimental` namespace, to indicate that these
```
Comment (vepadulano) [inline tree/dataframe/src/RDataFrame.cxx:1062]: Pinning this to a version gets outdated pretty quickly (as we are seeing already now), maybe I would change the wording to `Currently, ...` also in the similar sentence below

**Review by vepadulano — CHANGES_REQUESTED** [review]
Thank you! Some small suggestions

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 5]
```diff
@@ -695,10 +700,32 @@ public:
                             theVariationName);
    }
 
-   /// \brief Register systematic variations for an existing columns using auto-generated variation tags.
-   /// This overload of Vary takes a nVariations parameter instead of a list of tag names. Tag names
-   /// will be auto-generated as the sequence 0...nVariations-1.
-   /// See the documentation of the previous overload for more information.
+   /// \brief Register systematic variations for a single existing column using auto-generated variation tags.
+   /// \param[in] colName set of names of columns for which varied values are provided.
```
[the remark this answers is not in the document]
Comment (martamaja10) [reply tree/dataframe/inc/ROOT/RDF/RInterface.hxx:704]: good catch! thanks!

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 6]
```diff
@@ -695,10 +700,32 @@ public:
...
+   /// \param[in] expression a callable that evaluates the varied values for the specified columns. The callable can
+   ///            take any column values as input, similarly to what happens during Filter and Define calls. It must
+   ///            return an RVec of varied values, one for each variation tag, in the same order as the tags.
+   /// \param[in] inputColumns the names of the columns to be passed to the callable.
+   /// \param[in] nVariations number of variations returned by the expression. The corresponding tags will be "0", "1",
+   /// etc. \param[in] variationName a generic name for this set of varied values, e.g. "ptvariation".
```
[the remark this answers is not in the document]
Comment (martamaja10) [reply tree/dataframe/inc/ROOT/RDF/RInterface.hxx:710]: that must have been clang

File: tree/dataframe/src/RDataFrame.cxx (hunk 4 above)
Comment (martamaja10) [reply tree/dataframe/src/RDataFrame.cxx:1062]: makes sense!

**Review by enirolf — APPROVED** [review]
This is great, thanks! I've left a few small comments (mostly cosmetic)

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 7]
```diff
@@ -695,10 +700,34 @@ public:
...
+   ///   df.Vary("pt", [] (double pt) { return RVecD{pt*0.9, pt*1.1}; }, 2)
+   ///     .Histo1D("x");
+   ///
+   /// auto hx = ROOT::RDF::Experimental::VariationsFor(nominal_hx);
+   /// hx["nominal"].Draw();
+   /// hx["x:0"].Draw("SAME");
+   /// hx["x:1"].Draw("SAME");
+   /// ~~~
+   ///
+   /// See the documentation of another Vary() overload for more information.
```
[the remark this answers is not in the document]
Comment (martamaja10) [reply tree/dataframe/inc/ROOT/RDF/RInterface.hxx:730]: doxygen magically(?) assigns Vary() to the desired overload number 8 (so I will leave it like that for simplicity of reading), but thanks for the \sa suggestion!

**Review by vepadulano — APPROVED** [review]
Thank you!

**Pull request merged**